src/main/java/net/fabricmc/mappingio/tree/MappingCollectionView.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2025 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.mappingio.tree;
+
+import java.util.Collection;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.fabricmc.mappingio.tree.MappingTreeView.ClassMappingView;
+import net.fabricmc.mappingio.tree.MappingTreeView.ElementMappingView;
+import net.fabricmc.mappingio.tree.MappingTreeView.FieldMappingView;
+import net.fabricmc.mappingio.tree.MappingTreeView.MethodArgMappingView;
+import net.fabricmc.mappingio.tree.MappingTreeView.MethodMappingView;
+import net.fabricmc.mappingio.tree.MappingTreeView.MethodVarMappingView;
+
+/**
+ * A {@link Collection}-based view of element mappings present in a mapping tree.
+ *
+ * <p>Contrary to what's defined in {@link Collection}'s Javadocs, the {@code add}
+ * methods here do not guarantee adding a passed element into the collection,
+ * instead its data may be merged into a compatible existing entry in case and such that
+ * {@link #containsCompatible(Object)} returns {@code true} for the passed element.
```

Comment:
This should probably clarify what entries are compatible (mentioning src name/desc and the other "key" properties like lv index)